src/log/log.c

```diff
@@ -1904,9 +1904,9 @@ __log_has_hole(WT_SESSION_IMPL *session, WT_FH *fh, wt_off_t log_size,
 			 * hole.  We can do some simple data verification to
 			 * ensure that this part of the file is not corrupted.
 			 */
-			for (p = buf; remainder > 0;
-			     remainder -= (wt_off_t)rdlen, p += rdlen) {
-				rdlen = WT_MIN(WT_LOG_ALIGN, (size_t)remainder);
+			for (p = buf, buf_left = rdlen; buf_left > 0;
+			     buf_left -= rdlen, p += rdlen) {
+				rdlen = WT_MIN(WT_LOG_ALIGN, buf_left);
```

Comment:
I've been reading through this and thinking about this and understand it better. Although it isn't changed in this diff, I'm thinking `WT_LOG_ALIGN` should be `log->allocsize`. That is the unit that log records are aligned on. So if someone is using a non-standard log alignment, it should use that. (Might be good to add a non-standard log allocation size in the corruption tests too.)